xmeffects.js

```diff
@@ -65,14 +65,35 @@ function eff_t0_4(ch, data) {  // vibrato
 }
 
 function eff_t1_4(ch) {  // vibrato
-  ch.periodoffset = Math.sin(ch.vibratopos * Math.PI / 32) * ch.vibratodepth;
+  ch.periodoffset = getVibratoDelta(ch.vibratotype, ch.vibratopos) * ch.vibratodepth;
   if (isNaN(ch.periodoffset)) {
     console.log("vibrato periodoffset NaN?",
         ch.vibratopos, ch.vibratospeed, ch.vibratodepth);
     ch.periodoffset = 0;
   }
-  ch.vibratopos += ch.vibratospeed;
-  ch.vibratopos &= 63;
+  // only updates on non-first ticks
+  if (player.cur_tick > 0) {
+    ch.vibratopos += ch.vibratospeed;
+    ch.vibratopos &= 63;
+  }
```

Comment:
The [OpenMPT source](https://source.openmpt.org/browse/openmpt/trunk/OpenMPT/soundlib/Sndmix.cpp) says this about advancing vibrato position
> IT updates on every tick, unless "old effects" are enabled (in this case it only updates on non-first ticks like other trackers)

Search for `Advance vibrato position` in the file above to check it out